### PR TITLE
Add test case for parallel command deduplication using mixed clients [KVL-1090]

### PR DESCRIPTION
**Pull request by nicu-da**
- Use CommandService and CommandSubmissionService in parallel to test command deduplication

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala [hunk 1]
```diff
@@ -63,33 +59,44 @@ class AppendOnlyCommandDeduplicationParallelIT extends LedgerTestSuite {
...
   private def runTestWithSubmission[T](
       ledger: ParticipantTestContext,
       party: Party,
-      request: T,
-      submitRequestAndGetStatus: T => Future[Code],
+      submitRequestAndGetStatus: () => Future[Code],
   )(implicit
       ec: ExecutionContext
   ) = {
     val numberOfParallelRequests = 10
```
Comment (fabiotudone-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala:95]: I think it's clearer if this knob controlled by the caller or is at least a suite-scoped constant.

**Review by fabiotudone-da — APPROVED** [review]

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala [hunk 2]
```diff
@@ -109,6 +116,36 @@ class AppendOnlyCommandDeduplicationParallelIT extends LedgerTestSuite {
     }
   }
 
+  private def submitRequest(
+      ledger: ParticipantTestContext,
+      party: client.binding.Primitive.Party,
+  ) = {
```
Comment (hubert-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala:122]: braces :D

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala [hunk 3]
```diff
@@ -109,6 +116,36 @@ class AppendOnlyCommandDeduplicationParallelIT extends LedgerTestSuite {
...
+        _.commands.deduplicationPeriod := DeduplicationPeriod.DeduplicationDuration(
+          deduplicationDuration.asProtobuf
+        )
+      )
+  }
+
+  private def submitAndWaitRequest(
+      ledger: ParticipantTestContext,
+      party: binding.Primitive.Party,
+  ) = {
```
Comment (hubert-da) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala:138]: braces :D

**Review by hubert-da — APPROVED** [review]

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala (hunk 2 above)
Comment (nicu-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala]: Done

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala (hunk 1 above)
Comment (nicu-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala]: I extracted it into the suite scope

**Pull request merged**